spec/an_metadata_update_helper_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'tmpdir'
+require_relative './spec_helper'
+
+describe Fastlane::Helper::StandardMetadataBlock do
+  it 'strips any trailing newline when generating the block for a single-line input' do
+    Dir.mktmpdir do |dir|
+      input = "Single line message with new line\n"
+
+      # Generate the input file to convert to .pot block
+      input_path = File.join(dir, 'input')
+      File.write(input_path, input)
+      # Ensure the input has only one line
+      expect(File.read(input_path).lines.count).to eq 1
+
+      # Write the .pot block in a StringIO to bypass the filesystem and have
+      # faster tests
+      output_io = StringIO.new
+      described_class.new('any-key', input_path).generate_block(output_io)
+
+      # Ensure the output matches the expectation: the trailing new line has
+      # been stripped.
+      #
+      # Note that the final new line is intentional. It's part of the
+      # formatting at the time of writing.
+      expect(output_io.string).to eq <<~EXP
+        msgctxt "any-key"
+        msgid "Single line message with new line"
+        msgstr ""
+
+      EXP
+    end
+  end
+
+  it 'does not strip a trailing new line when generating the block for a multi-line input' do
```

Comment:
Notice "_does not_ strip..." and the trailing `\n` in the last line. This is the current behavior (e.g.: [here](https://github.com/wordpress-mobile/WordPress-iOS/blob/2386060a1b9990a9a5d170b8e0877f3aa4f98b3a/WordPress/Resources/AppStoreStrings.po#L23-L34)) and, unlike the one this PR aims to fix in the single-line instance, it doesn't generate a validation error. 

Personally, I think we should strip that final `\n`, but I don't want to do it in the context of this PR and without discussing it.

Reply:
Yeah it might be fine to remove the trailing `\n` at the end of the _last_ line but I'm not sure either of the consequences there indeed, let's keep it for now as it doesn't do any harm.